From 13a4bfdff0a2979e358ed037a204904b87418337 Mon Sep 17 00:00:00 2001
From: Harald Sitter <sitter@kde.org>
Date: Mon, 24 Oct 2022 13:15:48 +0200
Subject: [PATCH] disable automatic portal launching

in Qt6 (and the KDE patch collection for Qt 5) genericunixservices will
internally attempt to probe the portal tech early on in the app life
cycle. this causes the protal system to launch app and then crash
because we aren't actually providing a fully functional session. instead
opt out of this altogether
---
 src/daemon/Greeter.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/daemon/Greeter.cpp b/src/daemon/Greeter.cpp
index 878576b..2e15446 100644
--- a/src/daemon/Greeter.cpp
+++ b/src/daemon/Greeter.cpp
@@ -188,6 +188,8 @@ namespace SDDM {
             //some themes may use KDE components and that will automatically load KDE's crash handler which we don't want
             //counterintuitively setting this env disables that handler
             env.insert(QStringLiteral("KDE_DEBUG"), QStringLiteral("1"));
+            // Qt internally may load the xdg portal system early on, prevent this, we do not have a functional session running.
+            env.insert(QStringLiteral("QT_NO_XDG_DESKTOP_PORTAL"), QStringLiteral("1"));
             m_auth->insertEnvironment(env);
 
             // log message
-- 
2.38.0

